src/Match2Cases.jl

```diff
@@ -263,48 +275,76 @@ function make_next(
+    end
+    succ
 end
-function remove(action::BoundTestPattern, sense::Bool, code::CodePoint)::CodePoint
-    cases = map(c -> remove(action, sense, c), code.cases)
-    return with_cases(code, cases)
+function remove(action::BoundTestPattern, sense::Bool, code::CodePoint, state::BinderState)::CodePoint
+    cases = map(c -> remove(action, sense, c, state), code.cases)
+    return CodePoint(cases)
 end
 
-function remove(action::BoundTestPattern, sense::Bool, case::CasePartialResult)::CasePartialResult
-    with_pattern(case, remove(action, sense, case.pattern))
+function remove(action::BoundTestPattern, sense::Bool, case::CasePartialResult, state::BinderState)::CasePartialResult
+    with_pattern(case, remove(action, sense, case.pattern, state))
 end
-function remove(action::BoundFetchPattern, case::CasePartialResult)::CasePartialResult
-    with_pattern(case, remove(action, case.pattern))
+function remove(action::BoundFetchPattern, case::CasePartialResult, state::BinderState)::CasePartialResult
+    with_pattern(case, remove(action, case.pattern, state))
 end
 
 #
 # Remove the given action from a pattern.
 #
-remove(action::BoundFetchPattern, pattern::BoundPattern)::BoundPattern = pattern
-remove(action::BoundTestPattern, sense::Bool, pattern::BoundPattern)::BoundPattern = pattern
-function remove(action::BoundFetchPattern, pattern::BoundFetchPattern)::BoundPattern
+remove(action::BoundFetchPattern, pattern::BoundPattern, state::BinderState)::BoundPattern = pattern
+remove(action::BoundTestPattern, sense::Bool, pattern::BoundPattern, state::BinderState)::BoundPattern = pattern
+function remove(action::BoundFetchPattern, pattern::BoundFetchPattern, state::BinderState)::BoundPattern
     return (action == pattern) ? BoundTruePattern(pattern.location, pattern.source) : pattern
 end
-function remove(action::BoundFetchPattern, pattern::Union{BoundAndPattern,BoundOrPattern})::BoundPattern
-    subpatterns = collect(BoundPattern, map(p -> remove(action, p), pattern.subpatterns))
+function remove(action::BoundFetchPattern, pattern::Union{BoundAndPattern,BoundOrPattern}, state::BinderState)::BoundPattern
+    subpatterns = collect(BoundPattern, map(p -> remove(action, p, state), pattern.subpatterns))
     return (typeof(pattern))(pattern.location, pattern.source, subpatterns)
 end
-function remove(action::BoundTestPattern, sense::Bool, pattern::BoundTestPattern)::BoundPattern
+function remove(action::BoundTestPattern, sense::Bool, pattern::BoundTestPattern, state::BinderState)::BoundPattern
     return (action == pattern) ? BoundBoolPattern(pattern.location, pattern.source, sense) : pattern
 end
-function remove(action::BoundTestPattern, sense::Bool, pattern::Union{BoundAndPattern,BoundOrPattern})::BoundPattern
-    subpatterns = collect(BoundPattern, map(p -> remove(action, sense, p), pattern.subpatterns))
+function remove(action::BoundEqualValueTestPattern, sense::Bool, pattern::BoundEqualValueTestPattern, state::BinderState)::BoundPattern
```

Comment:
could you add a comment on what `sense` is here and what it does? it's not 100% obvious to me from reading the code

Reply:
Done.